src/pages/LogIn/index.tsx

```diff
@@ -1,14 +1,44 @@
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { useNavigate } from 'react-router-dom';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect, useMemo } from 'react';
+import { logInSchema } from '../../schemas';
 import {
   Button, InputField, Logo, Paragraph, PhoneSection, Title, Subtitle,
 } from '../../components/common';
-import { usernameSelector } from '../../state/reducers/userReducer';
+import { usernameSelector, requestErrorsSelector, statusSelector } from '../../state/reducers/userReducer';
 import { RoutherPaths } from '../../constants/routerPaths';
+import { getFormErrors } from '../../utils/getFormErrors';
+import { LogInInputsType } from '../../types/userInputsTypes';
+import { logIn } from '../../state/actions/userActions';
+import { RequestStatus } from '../../constants/requestStatus';
+import useAuthentication from '../../hooks/useAuthentication';
 
 function LogIn() {
+  const dispatch = useDispatch();
   const navigate = useNavigate();
+  const isAuthenticated = useAuthentication();
+  const status = useSelector(statusSelector);
   const username = useSelector(usernameSelector);
+  const requestErrors = useSelector(requestErrorsSelector);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(logInSchema) });
+
+  const isRequestLoading = useMemo(() => status === RequestStatus.PENDING, [status]);
+
+  const onSubmit: SubmitHandler<LogInInputsType> = async (data: LogInInputsType) => {
+    await dispatch(logIn(data)).unwrap();
+    navigate(RoutherPaths.HOME);
+  };
+
+  useEffect(() => {
```

Comment:
```
if(isAuthenticated) {
  ...
}
```
without useEffect should fix all redirection stuff, and you could remove line 36

Reply:
Actually I'll get a warning from react.dev.tools if I remove the useEffect hook, it recommends you to always call navigate() inside a useEffect()